### DISCUSSION TITLE
Is it necessary to add prefault option in memory?

### DISCUSSION THREAD

**Discussion by yukiiiteru** [cloud-hypervisor/cloud-hypervisor#3186] [Q&A] [answered]
I've noticed that `prefault` argument is available in memory manager, but it seems that this argument can only be used by SGX and restore as an option.

I think `prefault`, in other words, `MAP_POPULATE` is very useful not only in SGX or restore, but also in normal memory allocation.

With this flag, vmm can pre-alloc page tables of vm memory without subsequent page faults, although it can make memory allocation (`mmap`) slower.

So, is it necessary to add `prefault` option in memory?

**Comment by sboeuf** [comment, marked as answer]
Yes you're right, so far we only provide the option for SGX and restore use cases. But if you have a specific use case in mind for which having the memory populated upfront might help, it shouldn't be too hard to add it to the normal memory (the code is mostly there, it's really a matter of exposing the option to the user and connecting the dots).
